### PR TITLE
Add gfastats

### PR DESCRIPTION
**Pull request by gallardoalba**

### CODE REVIEW

**Comment by bgruening** [comment]
Conda package is in 10min available ... and container as well.

File: tools/gfastats/gfastats.xml [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+                <option value="manipulation">Genome assembly manipulation</option>
+                <option value="statistics">Summary statistics generation</option>
+            </param>
+            <when value="manipulation">
+                <param argument="--swiss-army-knife" type="data"
+                    format="txt" label="SAK input file" optional="true"
+                    help="Set of instructions provided as an ordered list"/>
+                <conditional name="output_condition">
+                    <param argument="--out-format" type="select" 
+                        label="Output format" help="Outputs selected sequences. If more than the extension is provided the output is written to the specified file (e.g. out.fasta.gz).">
```
Comment (bgruening) [inline tools/gfastats/gfastats.xml:101]: I do not understand the help text. Is that relevant to the user?

File: tools/gfastats/gfastats.xml [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+                <param argument="--exclude-bed" type="data" format="bed" optional="true"
+                    label="Exclude specific intervals"
+                    help="Exclude a subset of headers or coordinates in 0-base bed format. It can be conmbined with --include-bed Optional"/>
+            </when>
+            <when value="false"/>
+        </conditional>
+        <conditional name="mode_condition">
+            <param name="selector" type="select" label="Tool mode">
+                <option value="manipulation">Genome assembly manipulation</option>
+                <option value="statistics">Summary statistics generation</option>
```
Comment (bgruening) [inline tools/gfastats/gfastats.xml:93]: is stats maybe the preferred option, the most used option? then add it as first option

File: tools/gfastats/gfastats.xml [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                            label="Expected genome size" help="Estimated genome size. This parameter is optional, but required for NG* statistics."/>
+                    </when>
+                    <when value="sequence">
+                        <param argument="--out-sequence" type="boolean" truevalue="--out-sequence" falsevalue="" checked="false"
+                            label="Report actual sequence" help="It reports also the actual sequence"/>
+                    </when>
+                </conditional>
+            </when>
+        </conditional>
+        <param argument="--tabular" type="boolean" truevalue="--tabular" falsevalue="" checked="false"
```
Comment (bgruening) [inline tools/gfastats/gfastats.xml:156]: where does this param play a role in the outputs?

File: tools/gfastats/macros.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                publisher = {GitHub},
+                journal = {GitHub repository},
+                url = {https://github.com/vgl-hub/gfastats}}
+            </citation>
+        </citations>
+    </xml>
+    <xml name="length_macro">
+        <param argument="--line-length" type="integer" min="0" max="80" value=""
+            optional="true" label="Especify line length"
+            help="Specifies line length in when output format is fasta. Default has no line breaks."/>
```
Comment (bgruening) [inline tools/gfastats/macros.xml:30]: "in when"?

File: tools/gfastats/gfastats.xml (hunk 2 above)
Comment (gallardoalba) [reply tools/gfastats/gfastats.xml:93]: Yes, probably it is the preferred one.

File: tools/gfastats/gfastats.xml (hunk 1 above)
Comment (gallardoalba) [reply tools/gfastats/gfastats.xml:101]: Not at all.

Comment (gallardoalba) [reply tools/gfastats/gfastats.xml:101]: It does explain that when adding a prefix, then it generates an output file; otherwise, it is sent to standard output.

File: tools/gfastats/gfastats.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    <description>the swiss army knife for genome assembly</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="biotools"/>
+    <version_command>gfastats --version</version_command>
+    <command detect_errors="exit_code"><![CDATA[
+        gfstats
+        $input_file
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/gfastats/gfastats.xml]: Yes, it was fixed in a previous PR.

File: tools/gfastats/gfastats.xml (hunk 3 above)
Comment (gallardoalba) [reply tools/gfastats/gfastats.xml:156]: Yes, indeed by default the output format should be text; I'll set it as default.

File: tools/gfastats/gfastats.xml (hunk 1 above)
Comment (bgruening) [reply tools/gfastats/gfastats.xml:101]: But this information is not relevant for a Galaxy user, left?

**Comment by gallardoalba** [comment]
All the recommendation has been fixed; in addition, I included two additional parameters (--sort and --homopolymer-compress) which have been included in the last update.

**Review by bgruening — APPROVED** [review]
